### PR TITLE
fix #12150 - prevent crash trying to quit while modal dialog shown

### LINKED ISSUE #12150: [MU4 Issue] Holding Cmd/Ctr+Q to quit makes multiple copies of "Do you want to save cahnges.."dialog (and leads to crash on Windows)
**To Reproduce**
1. Open a score 
2. Make some changes
3. Press and hold Cmd+Q (Ctr+Q on Win/Lin) - **it makes multiple copies of "Do you want to save changes.."window**

**Mac/Lin**

https://user-images.githubusercontent.com/90187801/175382687-1b685711-90de-42b1-a840-a11567de2800.mov

**Win**

https://user-images.githubusercontent.com/90187801/175382738-fd7d6964-abbf-483d-8212-2352b0e750c8.mp4

**Platform information**
macOS, Windows, Linux

**Additional context**
On **Windows** it leads to **crash**

Doesn't occur in **MU3**

### PR DESCRIPTION
**Pull request by wizofaus**
Resolves: [#12150](https://github.com/musescore/MuseScore/issues/12150)

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: src/appshell/internal/applicationactioncontroller.cpp [hunk 1]
```diff
@@ -229,3 +230,9 @@ void ApplicationActionController::checkForUpdate()
 
     interactive()->info(trc("appshell", "No update available"), "");
 }
+
+bool ApplicationActionController::canReceiveAction(const mu::actions::ActionCode& code) const
+{
+    auto focus = QGuiApplication::focusWindow();
+    return !focus || focus->modality() == Qt::WindowModality::NonModal;
+}
```
Comment (wizofaus) [inline src/appshell/internal/applicationactioncontroller.cpp:238]: I don't think there are any commands this controller can handle that make sense when there's a modal popup shown

**Comment by DmitryArefiev** [comment]
Tested #12150 on Win10, Mac12, LinuxMint 20.2 - **FIXED**

**Comment by DmitryArefiev** [comment]
@RomanPudashkin Please review

**Comment by wizofaus** [comment]
Is there anything else I need to do here? Seems to be have been sitting here a while...

**Review by Eism — APPROVED** [review]

**Pull request merged**